runtimes/huggingface/tests/test_common.py

```diff
@@ -210,3 +213,63 @@ def test_pipeline_checks_for_eos_and_pad_token(
     m = load_pipeline_from_settings(hf_settings, model_settings)
 
     assert m._batch_size == expected_batch_size
+
+
+@pytest.mark.parametrize(
+    "inference_kwargs1, inference_kwargs2, expected",
+    [
+        (
+            {"max_length": 20},
+            {"max_length": 10},
+            True,
```

Comment:
It is unclear what expected means given the test case, i suggest to refactor a bit to make it more clearer. It might be just you just need to assert the number of tokens in each request is as follows expected (effectively converting it to 2 test cases)

Reply:
Updated this into 2 test cases. Also asserting the number of predicted tokens are the expected number of tokens